src/js_sync.hpp

```diff
@@ -404,7 +405,11 @@ void SyncClass<T>::populate_sync_config(ContextType ctx, ObjectType realm_constr
         }
 
         std::string raw_realm_url = Object::validated_get_string(ctx, sync_config_object, "url");
-
+        if (shared_user->token_type() == SyncUser::TokenType::Admin) {
+            static std::regex tilde("/~/");
+            raw_realm_url = std::regex_replace(raw_realm_url, tilde, "/__auth/");
```

Comment:
Unless the identity is not deserialized by object store when it is an admin token, we could rely here on the identity info in `shared_user` instead of hard-coding `__auth`.